src/algebra/ordered_monoid.lean

```diff
@@ -49,10 +49,38 @@ class ordered_add_comm_monoid (α : Type*) extends add_comm_monoid α, partial_o
 
 attribute [to_additive] ordered_comm_monoid
 
+/-- A linearly ordered additive commutative monoid. -/
+@[protect_proj, ancestor ordered_add_comm_monoid linear_order]
+class linear_ordered_add_comm_monoid (α : Type*)
+  extends linear_order α, ordered_add_comm_monoid α :=
+(lt_of_add_lt_add_left := λ x y z, by {
+  apply imp_of_not_imp_not,
+  intro h,
+  apply not_lt_of_le,
+  apply add_le_add_left,
+  -- type-class inference uses `a : linear_order α` which it can't unfold, unless we provide this!
+  -- `lt_iff_le_not_le` gets filled incorrectly with `autoparam` if we don't provide that field.
+  letI : linear_order α := by refine { le := le, lt := lt, lt_iff_le_not_le := _, .. }; assumption,
+  exact le_of_not_lt h })
+
+/-- A linearly ordered commutative monoid. -/
+@[protect_proj, ancestor ordered_comm_monoid linear_order, to_additive]
+class linear_ordered_comm_monoid (α : Type*)
+  extends linear_order α, ordered_comm_monoid α :=
+(lt_of_mul_lt_mul_left := λ x y z, by {
+  apply imp_of_not_imp_not,
+  intro h,
+  apply not_lt_of_le,
+  apply mul_le_mul_left,
+  -- type-class inference uses `a : linear_order α` which it can't unfold, unless we provide this!
```

Comment:
I'm pretty sure I wrote this comment - where was it copied from?

Reply:
A few lines below, where `linear_ordered_comm_monoid_with_zero` is defined.

Reply:
Does it now let you remove the other copy, since that class now inherits this one?

Reply:
I think the only thing that broke without that there was `zero_le_one'`, and that doesn't break now... so fingers crossed